src/runtime/recipe/particle.ts

```diff
@@ -223,7 +218,27 @@ export class Particle {
     this._unnamedConnections.splice(idx, 1);
   }
 
-  addSlotConnection(name) {
+  addSlotConnection(name: string) {
+    assert(!(name in this._consumedSlotConnections));
+    const slotConn = new SlotConnection(name, this);
+    this._consumedSlotConnections[name] = slotConn;
+
+    const slotSpec = this.getSlotSpecByName(name);
+    if (slotSpec) {
```

Comment:
Done :) 

There was a similar check in manifest.ts that throws a ManifestError if the slot connection doesn't exist in the spec. This error is useful because the message is more informative. I kept both this ManifestError and the assert inside the addSlotConnection method, but I moved the ManifestError up to before the slot connection is created to ensure that the ManifestError is thrown instead of the assert when parsing the manifest.